etna/transforms/decomposition/change_points_segmentation.py

```diff
@@ -0,0 +1,147 @@
+        change_point_model:
+            model to get change points
+        change_point_model_predict_params:
+            params for ``change_point_model.predict`` method
+        """
+        self.in_column = in_column
+        self.out_column = out_column
+        self.intervals: Optional[List[TTimestampInterval]] = None
+        self.ruptures_change_point_model = RupturesChangePointsModel(
+            change_point_model=change_point_model, **change_point_model_predict_params
+        )
+
+    def _fill_per_interval(self, series: pd.Series) -> pd.Series:
+        """Fill values in resulting series."""
+        if self.intervals is None:
+            raise ValueError("Transform is not fitted! Fit the Transform before calling transform method.")
+        result_series = pd.Series(index=series.index)
+        for k, interval in enumerate(self.intervals):
+            tmp_series = series[interval[0] : interval[1]]
+            if tmp_series.empty:
+                continue
+            result_series[tmp_series.index] = k
+        return result_series.astype(int).astype("category")
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Split df to intervals.
+
+        Parameters
+        ----------
+        df:
+            one segment dataframe
+
+        Returns
+        -------
+        df:
+            df with new column
+        """
+        series = df[self.in_column]
+        result_series = self._fill_per_interval(series=series)
+        df.loc[:, self.out_column] = result_series
+        return df
+
+    def fit(self, df: pd.DataFrame) -> "_OneSegmentChangePointsSegmentationTransform":
+        """Fit _OneSegmentChangePointsSegmentationTransform: find change points in ``df`` and build intervals.
+
+        Parameters
+        ----------
+        df:
+            one segment dataframe indexed with timestamp
+
+        Returns
+        -------
+        :
+
+        Raises
+        ------
+        ValueError
+            If series contains NaNs in the middle
+        """
+        self.intervals = self.ruptures_change_point_model.get_change_points_intervals(df=df, in_column=self.in_column)
+        return self
+
+    def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Do nothing in this case.
+
+        Parameters
+        ----------
+        df:
+            one segment dataframe
+
+        Returns
+        -------
+        df:
+            one segment dataframe
+        """
+        return df
+
+
+class ChangePointsSegmentationTransform(PerSegmentWrapper, FutureMixin):
+    """ChangePointsSegmentationTransform make label encoder to change points.
+
+    Warning
+    -------
+    This transform can suffer from look-ahead bias. For transforming data at some timestamp
+    it uses information from the whole train part.
+    """
+
+    def __init__(
+        self,
+        in_column: str,
+        change_point_model: BaseEstimator,
```

Comment:
This transform should get the instance of BaseChangePointsModelAdapter instead of  change_point_model and  **change_point_model_predict_params